angular/src/pipes/ellipsis.pipe.ts

```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+  name: "ellipsis",
+})
+export class EllipsisPipe implements PipeTransform {
+  transform(value: string, limit = 25, completeWords = false, ellipsis = "...") {
```

Comment:
What do you think about limit being the character limit vs the elipses-start-point? I think it's more intuitive to say "I have room for 25 characters" than to say "I want to start elipses at 22". Of course, this all depends on font. Any way we can calculate that instead?

Reply:
Ah yes, that totally makes sense! I'll change it to that way instead!

Reply:
I did a bunch of experimenting with this to see if it could be a bit smarter and return the correct string length based on the width of the containing element, but the solution I came up with felt dubious, so I'm sticking with doing it this more straightforward way 🙂 

I did update the original calculations to take into account the length of the ellipsis so the total length of the string returned accounts for it.